src/index/IndexImpl.cpp

```diff
@@ -543,27 +541,23 @@ IndexImpl::createPermutationPairImpl(const string& fileName1,
     lastLhs = triple[c1];
   }
   if (from < totalNumTriples) {
-    writer1.addRelation(currentRel.value(), buffer, distinctCol1);
-    writeSwitchedRel(&writer2, currentRel.value(), &buffer);
+    auto md1 = writer1.addRelation(currentRel.value(), buffer, distinctCol1);
+    auto md2 = writeSwitchedRel(&writer2, currentRel.value(), &buffer);
+    metaData1.add(md1);
+    metaData2.add(md2);
```

Comment:
This code duplication (identical) should be packed into a lambda. One can see the problems with duplication here:
I think the exchanging of the multiplicities is missing in this last block.